config/phpstan-baseline.neon

```diff
@@ -54,6 +54,12 @@ parameters:
 			count: 1
 			path: ../src/RuleSet/DeclarationBlock.php
 
+		-
+			message: '#^Parameters should have "string\|null" types as the only types passed to this method$#'
+			identifier: typePerfect.narrowPublicClassMethodParamType
+			count: 2
+			path: ../src/RuleSet/DeclarationBlock.php
+
```

Comment:
@oliverklee,

These warnings are unfathomable and look like bugs in the tool.  If I remove the call to `getRulesAssoc()` from https://github.com/MyIntervals/PHP-CSS-Parser/blob/1ce18b7810617b9795feb9276d5848d3d9896931/tests/ParserTest.php#L382 one of the warnings disappears.

But `getRulesAssoc()` is defined as accepting only `string` or `null`, and the caller there is passing a `string` constant.  Am I missing something?

Reply:
This warning comes from the PHPStan extension https://github.com/rectorphp/type-perfect, and this might (or might not) be a bug in that extension. I've just checked with their latest release 2.1.0 (which was released 4 days ago), but the warning still persists. I'll have a look into our code now.

Reply:
Yes, this looks like a bug in the tool to me. Somehow the tool gets hung up on the parameter being nullable.

Reply:
Ah, or this might also be just the way this rule works (or is intended to work):
https://github.com/rectorphp/type-perfect?tab=readme-ov-file#4-narrow-param-types

Reply:
If I understand the rule correctly, it is telling us that the parameter doesn't need to be nullable.

Reply:
Yes, it seems that the message is wrong and should be 'Parameters should have "string" types as the only types passed to this method'.  If I add an additional call to `getRulesAssoc(null)`, this resolves the warning.  But if I call `getRulesAssoc()` without an argument, the warning is still given, despite the default value of the parameter being `null`, so that does look like a bug in the tool.  Perhaps when tests are added, the warnings will disappear.